command/proxy.go

```diff
@@ -0,0 +1,1291 @@
+		return 1
+	}
+
+	if config.DisableIdleConnsAPIProxy {
+		proxyClient.SetMaxIdleConnections(-1)
+	}
+
+	if config.DisableKeepAlivesAPIProxy {
+		proxyClient.SetDisableKeepAlives(true)
+	}
+
+	apiProxyLogger := c.logger.Named("apiproxy")
+
+	// The API proxy to be used, if listeners are configured
+	apiProxy, err := cache.NewAPIProxy(&cache.APIProxyConfig{
+		Client:                  proxyClient,
+		Logger:                  apiProxyLogger,
+		EnforceConsistency:      enforceConsistency,
+		WhenInconsistentAction:  whenInconsistent,
+		UserAgentStringFunction: useragent.ProxyStringWithProxiedUserAgent,
+		UserAgentString:         useragent.ProxyString(),
+	})
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error creating API proxy: %v", err))
+		return 1
+	}
+
+	// Parse proxy cache configurations
+	if config.Cache != nil {
+		cacheLogger := c.logger.Named("cache")
+
+		// Create the lease cache proxier and set its underlying proxier to
+		// the API proxier.
+		leaseCache, err = cache.NewLeaseCache(&cache.LeaseCacheConfig{
+			Client:      proxyClient,
+			BaseContext: ctx,
+			Proxier:     apiProxy,
+			Logger:      cacheLogger.Named("leasecache"),
+		})
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error creating lease cache: %v", err))
+			return 1
+		}
+
+		// Configure persistent storage and add to LeaseCache
+		if config.Cache.Persist != nil {
+			if config.Cache.Persist.Path == "" {
+				c.UI.Error("must specify persistent cache path")
+				return 1
+			}
+
+			// Set AAD based on key protection type
+			var aad string
+			switch config.Cache.Persist.Type {
+			case "kubernetes":
+				aad, err = getServiceAccountJWT(config.Cache.Persist.ServiceAccountTokenFile)
+				if err != nil {
+					c.UI.Error(fmt.Sprintf("failed to read service account token from %s: %s", config.Cache.Persist.ServiceAccountTokenFile, err))
+					return 1
+				}
+			default:
+				c.UI.Error(fmt.Sprintf("persistent key protection type %q not supported", config.Cache.Persist.Type))
+				return 1
+			}
+
+			// Check if bolt file exists already
+			dbFileExists, err := cacheboltdb.DBFileExists(config.Cache.Persist.Path)
+			if err != nil {
+				c.UI.Error(fmt.Sprintf("failed to check if bolt file exists at path %s: %s", config.Cache.Persist.Path, err))
+				return 1
+			}
+			if dbFileExists {
+				// Open the bolt file, but wait to setup Encryption
+				ps, err := cacheboltdb.NewBoltStorage(&cacheboltdb.BoltStorageConfig{
+					Path:   config.Cache.Persist.Path,
+					Logger: cacheLogger.Named("cacheboltdb"),
+				})
+				if err != nil {
+					c.UI.Error(fmt.Sprintf("Error opening persistent cache: %v", err))
+					return 1
+				}
+
+				// Get the token from bolt for retrieving the encryption key,
+				// then setup encryption so that restore is possible
+				token, err := ps.GetRetrievalToken()
+				if err != nil {
+					c.UI.Error(fmt.Sprintf("Error getting retrieval token from persistent cache: %v", err))
```

Comment:
I think this behaviour makes sense, yeah. This code is copied from the old Agent behaviour (so that we match behaviours exactly). In short, if we can't reload our token from the persistent cache, we start anyway (and fetch a new token), which feels like a valid behaviour.